pkg/reconciler/common/testing/util.go

```diff
@@ -85,3 +89,92 @@ func AssertDeepEqual(t *testing.T, actual, expected interface{}) {
 	}
 	t.Fatalf("expected does not deep equal actual. \nExpected: %T %+v\nActual:   %T %+v", expected, expected, actual, actual)
 }
+
+func ResourceMatchWithPath(actual mf.Manifest, expectedManifestPath string) bool {
+	if expectedManifestPath == "" && len(actual.Resources()) == 0 {
+		return true
+	}
+	expected, err := mf.NewManifest(expectedManifestPath)
+	if err != nil {
+		return false
+	}
+	return ResourceMatch(actual, expected)
+}
+
+// ResourceMatch returns true if the resources in the actual manifest match the same resources in
+// the expected manifest, in terms of name, namespace, group and kind.
+func ResourceMatch(actual, expected mf.Manifest) bool {
+	// The resource match in terms of name, namespace, kind and group.
+	if len(actual.Filter(mf.Not(mf.In(expected))).Resources()) != 0 {
+		return false
+	}
+	if len(expected.Filter(mf.Not(mf.In(actual))).Resources()) != 0 {
+		return false
+	}
+	return true
+}
+
+func DeepMatchWithPath(actual mf.Manifest, expectedManifestPath string) bool {
+	if expectedManifestPath == "" && len(actual.Resources()) == 0 {
+		return true
+	}
+	expected, err := mf.NewManifest(expectedManifestPath)
+	if err != nil {
+		return false
+	}
+	return ResourceDeepMatch(actual, expected)
+}
+
+// ResourceDeepMatch returns true if the resources in the actual manifest match exactly the same resources in
+// the expected manifest.
+func ResourceDeepMatch(actual, expected mf.Manifest) bool {
```

Comment:
Can use lower case e.g. `resourceDeepMatch()` when the function can be private?

Reply:
will do